pkg/database/sync.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package database
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"sync"
+
+	pgx "github.com/jackc/pgx/v4"
+	"github.com/open-cluster-management/search-indexer/pkg/model"
+	"k8s.io/klog/v2"
+)
+
+const BATCH_SIZE = 500
+
+func SyncData(event model.SyncEvent, clusterName string) {
+	wg := &sync.WaitGroup{}
+	batch := &pgx.Batch{}
+	count := 0
+
+	// ADD
+	for _, resource := range event.AddResources {
+		json, _ := json.Marshal(resource.Properties)
+		batch.Queue("INSERT into resources values($1,$2,$3)", resource.UID, clusterName, string(json))
+		count++
+		if count == BATCH_SIZE {
+			wg.Add(1)
+			go sendBatch(*batch, wg)
```

Comment:
question here from @xrajesh: How do we know if something fails here and the data is not inserted? How is that handled? Will the collector know about the failure?

Reply:
We need to add error detection but I didn't get there yet.
In the old implementation we send errors back to the collector. We'll need to keep that.
Also in the old implementation we retry a few times just in case there's a transient network problem.

I'm thinking we can use errors channel where sendBatch() reports any errors.  We'll process the errors after `wg.Wait()`.